### PR TITLE
[CodeQuality] Resolve type on Collection<string, ClassName> docblock to correct target type when has indexBy

### PR DESCRIPTION
**Pull request by samsonasik**
@TomasVotruba ok, let's see if I can do it :)

per discussion at https://github.com/rectorphp/rector-doctrine/pull/361#pullrequestreview-2518615015

Closes https://github.com/rectorphp/rector-doctrine/pull/361

### CODE REVIEW

**Comment by samsonasik** [comment]
implemented 🎉 /cc @AlexeyKosov @yguedidi 

All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules-tests/CodeQuality/Rector/Property/ImproveDoctrineCollectionDocTypeInEntityRector/Source/Training.php [hunk 1]
```diff
@@ -4,7 +4,13 @@
...
+use Doctrine\ORM\Mapping as ORM;
+
 final class Training
 {
-
+    /**
+     * @ORM\Column(name="id", type="string")
+     * @ORM\Id
+     */
+    private string $id;
```
Comment (TomasVotruba) [inline rules-tests/CodeQuality/Rector/Property/ImproveDoctrineCollectionDocTypeInEntityRector/Source/Training.php:15]: Looks good 👍 

Could you add test fixture for attribute+ int type?

Comment (samsonasik) [reply rules-tests/CodeQuality/Rector/Property/ImproveDoctrineCollectionDocTypeInEntityRector/Source/Training.php:15]: sure, added integer type for annotation and attribute 👍 https://github.com/rectorphp/rector-doctrine/pull/362/commits/f84e8a5dab7a367b41d0d6c248d59f815b99c442

Comment (samsonasik) [reply rules-tests/CodeQuality/Rector/Property/ImproveDoctrineCollectionDocTypeInEntityRector/Source/Training.php:15]: also added more fixture for both attribute on current and target class https://github.com/rectorphp/rector-doctrine/pull/362/commits/724422732704f21b4a4fd1c9c218b470415d4cae

Comment (samsonasik) [reply rules-tests/CodeQuality/Rector/Property/ImproveDoctrineCollectionDocTypeInEntityRector/Source/Training.php:15]: also added constant fetch read on type target https://github.com/rectorphp/rector-doctrine/pull/362/commits/fc0a97466617a93d73172df2b00c07f890a953c3

**Comment by samsonasik** [comment]
@TomasVotruba ready to merge 👍

**Comment by samsonasik** [comment]
@TomasVotruba let's merge 👍

**Pull request merged**